CONSOLE-3883: Improve local plugin development docs (#13679)
openshift/console: frontend/packages/console-dynamic-plugin-sdk/README.md

**Important!** Make sure to provide valid JavaScript MIME type via the `Content-Type` response header
for all assets served by your plugin web server.

## Local plugin development

Clone Console repo and build the Bridge server by running `build-backend.sh` script.

Run the following commands to log in as `kubeadmin` user and start a local Bridge server instance.
The `-plugins` argument tells Bridge to force load your plugin upon Console application startup.
The `-i18n-namespaces` argument registers the corresponding i18n namespace for your plugin in Console.

```sh
oc login https://example.openshift.com:6443 -u kubeadmin -p example-password
source ./contrib/oc-environment.sh
# Note: the plugin web server URL should include a trailing slash
./bin/bridge -plugins foo-plugin=http://localhost:9001/ -i18n-namespaces=plugin__foo-plugin
```

To work with multiple plugins, provide multiple arguments to Bridge server:

```sh
./bin/bridge \
  -plugins foo-plugin=http://localhost:9001/ -i18n-namespaces=plugin__foo-plugin \
  -plugins bar-plugin=http://localhost:9002/ -i18n-namespaces=plugin__bar-plugin
```

> Once the Bridge server is running, start your plugin web server(s) and ensure that plugin assets can

Reviewer: ```suggestion
Once the Bridge server is running, start your plugin web server(s), and ensure that plugin assets can
```

be fetched via `/api/plugins/<plugin-name>` Bridge endpoint. For example, the following URLs should
provide the same content:

- http://localhost:9000/api/plugins/foo-plugin/plugin-manifest.json
- http://localhost:9001/plugin-manifest.json

> Open Console in web browser and inspect the value of `window.SERVER_FLAGS.consolePlugins` to see the

Reviewer: ```suggestion
Open the Console in your web browser and inspect the value of `window.SERVER_FLAGS.consolePlugins` to see the
```

> list of dynamic plugins which Console loads upon its startup. For local development, this should only

Reviewer: ```suggestion
list of dynamic plugins the Console loads at runtime. For local development, this should only
```

include plugin(s) listed via `-plugins` Bridge argument.

### Using local Console plugin SDK code

> If you need to make modifications to Console dynamic plugin SDK code and have them reflected in your

Reviewer: ```suggestion
If you need to make modifications to Console dynamic plugin SDK code and reflect them in your
```

plugin builds, follow these steps:

1. Make changes in Console repo. Run `yarn build` in `frontend/packages/console-dynamic-plugin-sdk`
   directory to rebuild plugin SDK files at `frontend/packages/console-dynamic-plugin-sdk/dist`.
2. Make sure your plugin's `package.json` dependencies refer to local plugin SDK files, for example:
```json
"@openshift-console/dynamic-plugin-sdk": "file:../openshift/console/frontend/packages/console-dynamic-plugin-sdk/dist/core",
"@openshift-console/dynamic-plugin-sdk-webpack": "file:../openshift/console/frontend/packages/console-dynamic-plugin-sdk/dist/webpack",
```

> 3. Refresh your plugin's `node_modules` whenever the local plugin SDK files are changed:

Reviewer: ```suggestion
3. Refresh your plugin's `node_modules` whenever you change local plugin SDK files:
```
